src/server/req_quejob.c

```diff
@@ -610,8 +610,8 @@ req_quejob(struct batch_request *preq)
 			return;
 		}
 		created_here = JOB_SVFLG_HERE;
+		free(namebuf);
```

Comment:
I suggest that you also move the declaration of namebuf to the top of this else block rather than at the top of the function, thereby minimizing the scope of the variable. Had this been done previously, the compiler would have caught the problem.